util/job_manager.rb

```diff
@@ -58,22 +58,22 @@ def perform
         @wrapper = wrapper
         log "avancement: #{message} #{avancement}"
         if avancement && cost
-          p = Result.get(self.uuid) || { 'graph' => [] }
-          p['graph'] = [] if !p.has_key?('graph')
-          p['csv'] = true if ask_restitution_csv
-          p['graph'] << { iteration: avancement, cost: cost, time: time }
-          p['result'] = solution if solution
+          p = Result.get(self.uuid) || { graph: [] }
+          p[:graph] ||= []
+          p[:csv] = true if ask_restitution_csv
+          p[:graph] << { iteration: avancement, cost: cost, time: time }
+          p[:result] = solution if solution
           Result.set(self.uuid, p)
         end
       }
 
       # Add values related to the current solve status
       p = Result.get(self.uuid) || {}
-      p['csv'] = true if ask_restitution_csv
-      p['result'] = result
-      if services_vrps.size == 1 && p && !p['result'].empty? && p['graph'] && !p['graph'].empty?
-        p['result'].first['iterations'] = p['graph'].last['iteration']
-        p['result'].first['elapsed'] = p['graph'].last['time']
+      p[:csv] = true if ask_restitution_csv
+      p[:result] = result
+      if services_vrps.size == 1 && p && !p[:result].empty? && p[:graph] && !p[:graph].empty?
```

Comment:
we can simplify this as follows I think
```suggestion
      if services_vrps.size == 1 && p && p[:result].any? && p[:graph]&.any?
```